### ISSUE TITLE
"cordova build ios" again overwrites CFBundleDisplayName in info.plist, removing the changes done when adding this plugin.

### ISSUE THREAD

**Issue by sachingargtech** [blakgeek/cordova-plugin-app-name#14]
Adding this plugin successfully changes values for both CFBundleDisplayName and CFBundleName in info.plist. 

But running "cordova build ios" after adding this plugin again overwrites CFBundleDisplayName with the default name value from app's config.xml, ignoring the plugin.xml. 

CFBundleName value does not change again, only CFBundleDisplayName value gets modified again. iOS only checks CFBundleDisplayName if it exists, so CFBundleName gets ignored by iOS.

**Comment by consciousimages** [comment]
Yes, exactly.  Confusingly, still an issue and defeats the purpose of using this plugin.  Android app names aren't the problem, iOS app names including spaces is -- at least for us.

**Comment by Amaury-T** [comment]
Same issue here... Have you found a solution?

**Comment by consciousimages** [comment]
No, I finally came to the conclusion that it’s a Cordova bug - it has to be as the results with and without the plugin are always pretty much the same. Occasionally (and seemingly randomly) the name would be correct but I could never get anything I tried to work consistently. In the end I more or less gave up.

**Comment by ZaLiTHkA** [comment]
Completely random thought: Does anyone know for sure if the Cordova CLI handles plugins synchronously or asynchronously during a build?

I'm just wondering if the order in which the plugins are listed might possibly affect this; such as perhaps some other plugin that also writes `CFBundleDisplayName` *after* this plugin has written it's value.

---

Although, having said that...

> Occasionally (and seemingly randomly) the name would be correct but I could never get anything I tried to work consistently.

This makes me think it might be handling them asynchronously, which means this might be some race condition with another plugin that's also updating the same value.

@sachingarg05, maybe you could try reproduce this in your environment with a fresh project and *only* `cordova-plugin-app-name` installed? I suspect this might show something useful.

**Comment by LyuGGang** [comment]
Same here 😂

**Comment by lovelyelfpop** [comment]
Any solutions?

**Comment by jurki** [comment]
In config.xml the name element has a "short" attribute where you can set the CFBundleDisplayName with unicode characters like this:

`<widget ...>`
`    <name short="Hello&#x2007;Cordova">HelloCordova</name>`
`</widget>`

I found this solution in the documentation:
https://cordova.apache.org/docs/en/9.x/config_ref/index.html#short-name

**Comment by sonicwong** [comment]
Just wrote a hook to replace "CFBundleDisplayName" in plist.
Anyone can merge this in plugin?
https://github.com/sonicwong/cordova-hook-ios-CFBundleDisplayName/blob/main/iosCFBundleDisplayName.js
